prism/src/main/java/org/apache/falcon/resource/AbstractEntityManager.java

```diff
@@ -161,8 +161,15 @@ protected static void checkColo(String colo) {
             Set<String> clusters = EntityUtil.getClustersDefined(entity);
             Set<String> colos = new HashSet<String>();
             for (String cluster : clusters) {
-                Cluster clusterEntity = EntityUtil.getEntity(EntityType.CLUSTER, cluster);
-                colos.add(clusterEntity.getColo());
+                try{
+                    Cluster clusterEntity = EntityUtil.getEntity(EntityType.CLUSTER, cluster);
+                    colos.add(clusterEntity.getColo());
+                } catch (EntityNotRegisteredException e){
+                    e.printStackTrace();
+                }
+            }
+            if (colos.size() == 0) {
```

Comment:
When an entity is scheduled on more than colo and gets deleted from one of them it throw's EntityNotFound exception and doesn't check for rest of the colo's.
I made this change so that it check in all the colo's from the entity definition.